R/abm.R

```diff
@@ -269,18 +280,21 @@ abm_table4 <- function(con = con_bib(), unit_code, analysis_start = abm_config()
 #' @import DBI dplyr tidyr purrr
 #' @export
 
-abm_table5 <- function(con = con_bib(), unit_code, analysis_start = abm_config()$start_year, analysis_stop = abm_config()$stop_year){
+abm_table5 <- function(con, unit_code, analysis_start = abm_config()$start_year, analysis_stop = abm_config()$stop_year){
 
-  # Get publication level data for selected unit (and filter on pub_year if given), relevant WoS doctypes only
-  orgdata <- abm_data(con = con, unit_code = unit_code) %>%
-    collect() %>%
-    filter(Publication_Type_WoS %in% c("Article", "Review") &
-             Publication_Year %in% analysis_start:(analysis_stop)
-            & !is.na(int))
+  # Get publication level data for selected unit, relevant WoS doctypes only
+  orgdata <- con %>%
+    tbl("masterfile") %>%
+    filter(Unit_code == unit_code &
+             Publication_Year >= analysis_start &
+             Publication_Year <= analysis_stop &
+             Publication_Type_WoS %in% c("Article", "Review") &
+             !is.na(int))
 
   # Duplicate rows so that publications are connected to all intervals they should belong to according to publication year
   orgdata3year <-
-    orgdata %>% 
+    orgdata %>%
+    collect() %>% 
     inner_join(sliding_intervals(analysis_start, analysis_stop, 3),
```

Comment:
This is not important really but I saw this package "funneljoin" which is now on CRAN which seems to have some nice join functions for time series data kept in data frames: https://hookedondata.org/introducing-the-funneljoin-package/

Reply:
Also, again not relevant here, but more stuff for tidy work with timeseries: https://tidyverts.org/

Reply:
Thanks for the links @mskyttner, both seem interesting! Maybe we can have a closer look later, when there is more time for refactoring.